agent/newrelic-install.sh

```diff
@@ -1471,25 +1486,49 @@ install_agent_here() {
 # file exists in the target directory.
 #
     if [ -n "${pi_inidir_cli}" -a ! -f "${pi_inidir_cli}/newrelic.ini" ]; then
-      if sed -e "s/REPLACE_WITH_REAL_KEY/${nrkey}/" "${ilibdir}/scripts/newrelic.ini.template" > "${pi_inidir_cli}/newrelic.ini"; then
+      # check if doing old style injection of just "newrelic.license" or if injecting values from env vars
+      if [ -n "${NR_CONFIG_WITH_ENVIRON}" ]; then
+        if php "${ilibdir}/newrelic-install-inject-envvars.php" "${ilibdir}/scripts/newrelic.ini.template" "${pi_inidir_cli}/newrelic.ini"; then
+          if [ "0" != "$?" ]; then
+            istat="failed"
+          fi
+        fi
+        if [ -z "${istat}" -a -z "${NR_INSTALL_SILENT}" ]; then
+          echo "      Install Status : ${pi_inidir_cli}/newrelic.ini created and environment variable configuration used"
+        fi
+      elif sed -e "s/REPLACE_WITH_REAL_KEY/${nrkey}/" "${ilibdir}/scripts/newrelic.ini.template" > "${pi_inidir_cli}/newrelic.ini"; then
         logcmd chmod 644 "${pi_inidir_cli}/newrelic.ini"
         if [ -z "${NR_INSTALL_SILENT}" ]; then
           echo "      Install Status : ${pi_inidir_cli}/newrelic.ini created"
         fi
       else
         istat="failed"
+      fi
+      istat="failed"
+      if [ -n "${istat}" ]; then
```

Comment:
if `istat` is set to `"failed"` on the line above, won't this statement always resolve to `true`? Can't we remove this conditional entirely in that case?

Reply:
Was this `istat="failed"` is intentional or was it accidentally copied from the `else` statement above?